version: '3'

services:
  cloudup_client:
    build: ./src
    ports:
      - "3031:3031"
    environment:
      - PORT=3031
      - FLAS_APP=app/__init__.py
      - FLASK_DEBUG=1
      - REDIS_URL="redis://redis"
      - REDIS_PORT=6379
      - DB_HOST=db.cloud-up-insight.com
      - DB_USER=root
      - DB_PASSWORD=cloudupusers
      - DB_NAME=cloudupdb
    volumes:
      - ./src:/opt/src
    command: >
      bash -c "./entryPoint.sh
      && mysql --host db.cloud-up-insight.com --port 3306 -u root -pcloudupusers -c "USE cloudupdb;""
    depends_on:
      - redis
    networks:
      - cloud-up

  inference_server:
    build: ./src
    environment:
      - REDIS_URL="redis://redis"
      - REDIS_PORT=6379
    command: ['python', 'app/models/InceptionV3/inception_inference_server.py']
    volumes:
      - ./src:/opt/src
    networks:
      - cloud-up
    depends_on:
      - cloudup_client
      - redis
  
  celery_worker:
    build: ./src
    command: ['celery', '-A', 'app.celeryapp:cloudup_celery_app', 'worker', '-l', 'info']
    volumes:
      - ./src:/opt/src
    networks:
      - cloud-up
    depends_on:
      - cloudup_client
    environment:
      - BROKER_URL=redis://redis:6379/0
  
  redis:
    image: redis:4.0.5-alpine
    command: ["redis-server", "--appendonly", "yes"]
    hostname: redis
    networks:
      - cloud-up
        
networks:
    cloud-up:
    